### ISSUE TITLE
Browser Policy Support

### ISSUE THREAD

**Issue by arajdon** [buttercup/buttercup-browser-extension#300]
For me as system administrator, this extension is missing a possibility to configure policies. Google Chrome (Firefox probably too) offers [extension policy support](https://www.chromium.org/administrators/configuring-policy-for-extensions). These can be set either as **preference** (user can change) or **enforced** (input disabled for user).

In my case, our company must not use cloud services like Google Drive, Dropbox or similar.

My suggestions for policies:
- disableGoogleDrive: boolean
- disableDropbox: boolean
- vaultAutoLockEnabled: boolean
- vaultAutoLockTime: int (seconds)
- showSaveDialog: int (0 = Never, 1= Always, 2 = When Vaults Unlocked)

**Comment by perry-mitchell** [comment]
@arajdon This is great, and I agree we need this. I'm not at all familiar with policies, so I appreciate any advice you have on the topic. I presume the policies are configured by some sys admin for a fleet of machines? And this info is available from the OS/browser?

Very interesting and I'd be happy to see this included.

**Comment by arajdon** [comment]
> I presume the policies are configured by some sys admin for a fleet of machines?
Right. These settings are deployed using any policy management system to either devices or directly to users in the company.

> And this info is available from the OS/browser?
For example, I've configured a whitelist for uBlock Origin by policy.

Windows Registry Setting: [HKEY_CURRENT_USER\SOFTWARE\Policies\Google\Chrome\3rdparty\extensions\cjpalhdlnbpafiamejdnhcphjbkeiagm\policy]
"adminSettings"="{\"netWhitelist\":\"domain.tld\\nanotherdomain.tld\"}"

Translates into:
Extension: cjpalhdlnbpafiamejdnhcphjbkeiagm
Policy Key: adminSettings
Policy Value: {"netWhitelist": "domain.tld anotherdomain.tld"}"

I guess that as soon as the browser recognizes the policy, the extension is able to read that. Values can be Integers, Strings and Booleans. To read all currently set policies open "chrome://policy".

If you have questions let me know, we're managing Chrome with such policies for roundabout 4000 devices.

**Comment by vikaspotluri123** [comment]
@perry-mitchell I did a bit of digging, and I think I found the page with information pertaining to this:

https://developer.chrome.com/extensions/manifest/storage

Here's uBlock's managed_storage.json as a reference:

**Comment by perry-mitchell** [comment]
Great stuff! Thanks for the links. I think adding policy support for the extension would be quite easy and definitely a good idea. Might be best to have a core policy module and then policy adapters per platform - the desktop app would most likely also benefit from this.
